### PR TITLE
[FPGA][NFC] Refactor [[intel::max_work_group_size()]] attribute tests

### PR DESCRIPTION
**Pull request by smanna12**
This patch updates test bases for [[intel::max_work_group_size()]] attribute:

1. separates AST and diagnostics to make it easier to read/follow-up.
2. removes duplicate test cases.
3. no compiler changes.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/test/SemaSYCL/intel-max-work-group-size-ast.cpp [hunk 1]
```diff
@@ -20,35 +19,73 @@ void foo() {
   });
 }
 
-#else // __SYCL_DEVICE_ONLY__
+#else  // __SYCL_DEVICE_ONLY__
```
[the remark this answers is not in the document]
Comment (smanna12) [reply clang/test/SemaSYCL/intel-max-work-group-size-ast.cpp]: Thanks @premanandrao for reviews. Yes, this is not needed. Removed

**Review by elizabethandrews — APPROVED** [review]

**Review by premanandrao — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you for reviews everyone.

**Comment by smanna12** [comment]
The failures are not related to my patch.

@intel/llvm-gatekeepers, this PR is ready for a merge. Thank you

**Comment by pvchupin** [comment]
Fails addressed at https://github.com/intel/llvm-test-suite/pull/1044

**Pull request merged**